### PR TITLE
Move BsRequest#with_reviews to the review model

### PR DESCRIPTION
**Pull request by bgeuken**
This method is querying reviews. Thus it belongs to the review model.
Since it's also just a query method I converted it into a scope.

### CODE REVIEW

File: src/api/app/models/bs_request.rb [hunk 1]
```diff
@@ -273,19 +273,11 @@ def self.sourcediff_has_shown_attribute?(sourcediff)
   #               or :target_project
   # @return [Array]  Array of Request objects
   def self.with_open_reviews_for(attributes)
-    with_reviews(attributes).where(state: 'new').rewhere('bs_requests.state': 'review').map(&:bs_request)
+    Review.with_attributes(attributes).where(state: 'new').rewhere('bs_requests.state': 'review').map(&:bs_request)
```
Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: [nitpick] Is the rewhere really necessary? Don't see where the bs_request.state can be set?

Comment (Ana06) [reply src/api/app/models/bs_request.rb]: I also think that `rewhere('bs_requests.state': 'review')` should be dropped. Even more, `where(state: 'new')` should IMO be a `rewhere` (to ensure we return the new ones) or more elegant:

```Ruby
Review.with(attributes.merge(state: 'new')).map(&:bs_request)
```

:bowtie:

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -273,19 +273,11 @@ def self.sourcediff_has_shown_attribute?(sourcediff)
...
   end
 
   def self.in_state_new(attributes)
```
Comment (Ana06) [inline src/api/app/models/bs_request.rb]: [nitpick] I would add a comment in that method similar to the one in [`self.with_open_reviews_for`](https://github.com/openSUSE/open-build-service/pull/6180/files#diff-d476b59ba5326de67e0c155e8e226d6fR274) to specify that this returns an Array. It is not evident/expected.

**Comment by bgeuken** [comment]
Dany and I wrote the two remaining request methods as scopes while working on the staging project code. I've added the commit now. That tackles the mapping issue @ChrisBr mentioned.

Regarding the review scope... I prefer to have it there over defining a method in the bs request model that essentially is a scope. Even if it's quite specific to these 2 request queries.

**Comment by bgeuken** [comment]
@Ana06 updated. Please have another look

File: src/api/app/models/staging/staging_project.rb [hunk 3]
```diff
@@ -16,7 +16,7 @@ def untracked_requests
     # The requests to review are requests which are not related to the staging project (unless they are also staged).
     # They simply need a review from the maintainers of the staging project.
     def requests_to_review
-      @requests_to_review ||= BsRequest.with_open_reviews_for(by_project: name)
+      @requests_to_review ||= BsRequest.with_open_reviews_for(by_project: name).includes(:reviews)
```
Comment (Ana06) [inline src/api/app/models/staging/staging_project.rb]: wouldn't it make sense to add this `includes` inside `with_open_reviews_for`? :thinking:

**Pull request merged**